build.sbt

```diff
@@ -1,197 +1,60 @@
-import sbtcrossproject.CrossPlugin.autoImport.{crossProject, CrossType}
-
-val Scala213 = "2.13.14"
-val Scala3 = "3.3.3"
-
-ThisBuild / crossScalaVersions := Seq(Scala213, Scala3)
-ThisBuild / scalaVersion := crossScalaVersions.value.last
-
-
-ThisBuild / githubWorkflowArtifactUpload := false
-
-val Scala213Cond = s"matrix.scala == '$Scala213'"
-
-def rubySetupSteps(cond: Option[String]) = Seq(
-  WorkflowStep.Use(
-    UseRef.Public("ruby", "setup-ruby", "v1"),
-    params = Map("ruby-version" -> "2.6.0"),
-    name = Some("Setup Ruby"),
-    cond = cond),
-
-  WorkflowStep.Run(
-    List(
-      "gem install saas",
-      "gem install jekyll -v 3.2.1"),
-    name = Some("Install microsite dependencies"),
-    cond = cond))
-
-ThisBuild / githubWorkflowBuildPreamble ++=
-  rubySetupSteps(Some(Scala213Cond))
-
-ThisBuild / githubWorkflowBuild := Seq(
-  WorkflowStep.Sbt(List("test", "mimaReportBinaryIssues")),
-
-  WorkflowStep.Sbt(
-    List("site/makeMicrosite"),
-    cond = Some(Scala213Cond)))
+ThisBuild / tlBaseVersion := "0.0" // your current series x.y
+
+ThisBuild / organization     := "cats"
+ThisBuild / organizationName := "Typelevel"
+ThisBuild / startYear        := Some(2024)
+ThisBuild / licenses         := Seq(License.MIT)
+ThisBuild / developers := List(
+  tlGitHubDev("ChristopherDavenport", "Christopher Davenport"),
+  tlGitHubDev("lenguyenthanh", "Thanh Le"),
+  tlGitHubDev("armanbilge", "Arman Bilge")
+)
 
-ThisBuild / githubWorkflowTargetTags ++= Seq("v*")
+// Use JDK 22
 ThisBuild / githubWorkflowJavaVersions := Seq(JavaSpec.temurin("17"))
 
-// currently only publishing tags
-ThisBuild / githubWorkflowPublishTargetBranches :=
-  Seq(RefPredicate.StartsWith(Ref.Tag("v")), RefPredicate.Equals(Ref.Branch("main")))
+ThisBuild / tlSonatypeUseLegacyHost := false
 
-ThisBuild / githubWorkflowPublishPreamble ++=
-  WorkflowStep.Use(UseRef.Public("olafurpg", "setup-gpg", "v3")) +: rubySetupSteps(None)
+ThisBuild / tlJdkRelease := Option(9)
 
-ThisBuild / githubWorkflowPublish := Seq(
-  WorkflowStep.Sbt(
-    List("ci-release"),
-    name = Some("Publish artifacts to Sonatype"),
-    env = Map(
-      "PGP_PASSPHRASE" -> "${{ secrets.PGP_PASSPHRASE }}",
-      "PGP_SECRET" -> "${{ secrets.PGP_SECRET }}",
-      "SONATYPE_PASSWORD" -> "${{ secrets.SONATYPE_PASSWORD }}",
-      "SONATYPE_USERNAME" -> "${{ secrets.SONATYPE_USERNAME }}")),
-  WorkflowStep.Use(UseRef.Public("christopherdavenport", "create-ghpages-ifnotexists", "v1")),
-  WorkflowStep.Sbt(
-    List("site/publishMicrosite"),
-    name = Some("Publish microsite"),
-    cond = Some(Scala213Cond)
-  )
-)
+val Scala3   = "3.4.2"
```

Comment:
IMHO, We should keep scala3 version `3.3.3` as it's the lts version. If We increase it to `3.4.2` users also have to bump their version to 3.4.2 which not ideal.